commons/che-core-commons-gwt/src/main/java/org/eclipse/che/ide/util/DebugUtils.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.util;
+
+import org.eclipse.che.commons.annotation.Nullable;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * Provides debug id for the class that implements {@link IDProvider}
+ *
+ * @author Vlad Zhukovskiy
+ */
+public final class DebugUtils {
+
+    /**
+     * Return generated debug id based on provided prefix.
+     *
+     * @param prefix
+     *         prefix
+     * @param provider
+     *         object that provides string id property
+     * @return generated id
+     */
+    public static String getDebugId(@Nullable String prefix, @NotNull IDProvider provider) {
+        StringBuilder sb = new StringBuilder();
+
+        if (prefix != null && !prefix.isEmpty()) {
+            sb.append(prefix).append('-');
+        }
+
+        sb.append(provider.getId());
+
+        return sb.toString();
+    }
+
+    private DebugUtils() {
```

Comment:
Do you follow `che-new` [codestyle](https://github.com/codenvy/codenvy-dev-resources/blob/master/che-codestyle/src/main/resources/che-codestyle-intellij-new.xml) ?